### PR TITLE
docs(*) improve docs for developer

### PR DESCRIPTION
**Pull request by mayocream**
### Summary

Enhance Kong repo's readme/guide to be more friendly to contributors and users.

- [x] Add Twitter badge and GitHub discussions link to README.md
- [x] Improve DEVELOPER.md by adding more sections about 
  - [x] Add installing OpenResty and LuaRocks by Linux packages.
  - [x] setup develop environment with Gojira.

### CODE REVIEW

**Comment by kikito** [comment]
I suggest splitting this PR into two prs:
* One for the `docs` changes
* Another one (of scope `chore`) just for the `.devcontainer`

**Comment by mayocream** [comment]
> I suggest splitting this PR into two prs:
> 
> * One for the `docs` changes
> * Another one (of scope `chore`) just for the `.devcontainer`

Thanks for noticing me! I'll follow your advice!

File: DEVELOPER.md [hunk 1]
```diff
@@ -287,9 +353,9 @@ Fedora:
         zlib-devel
 ```
 
-### OpenResty
+#### OpenResty
 
-We have a build script that makes it easy to pull and compile specific versions of the needed components of the OpenResty system.  Currently these include OpenResty 1.15.8.3, OpenSSl 1.1.1g, LuaRocks 3.3.1 and PCRE 8.44;  the exact versions can also be found on the [`.requirements`](https://github.com/Kong/kong/blob/master/.requirements) file of the main Kong repository.
+We have a build script that makes it easy to pull and compile specific versions of the needed components of the OpenResty system.  Currently these include OpenResty 1.19.9.1, OpenSSL 1.1.1m, LuaRocks 3.8.0 and PCRE 8.45;  the exact versions can also be found on the [`.requirements`](https://github.com/Kong/kong/blob/master/.requirements) file of the main Kong repository.
```
Comment (kikito) [inline DEVELOPER.md:358]: The version numbers will get out of date very quickly, I suggest removing them:

```suggestion
We have a build script that makes it easy to pull and compile specific versions of the needed components of the OpenResty system.  Their exact versions can be found on the [`.requirements`](https://github.com/Kong/kong/blob/master/.requirements) file.
```

**Review by kikito — APPROVED** [review]
Approved with one minor comment.

**Pull request merged**